File: 125-valid-palindrome/valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    s=s.toLowerCase();
    let str=""
    for(let i=0;i<s.length;i++){
        let char = s[i].charCodeAt(0);
        if((char>=97 && char<=122) || (char>=48 && char<=57)){
            str+=s[i];
        }
    }
    let left = 0,right = str.length-1;
    while(left<right){
        if(str[left]!=str[right]){
            return false;
        }
        left++;
        right--;
    }
    return true;
};

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder sb = new StringBuilder();
        for(int i = 0;i < s.length();i++){
            if(Character.isLetterOrDigit(s.charAt(i))){
                sb.append(Character.toLowerCase(s.charAt(i)));
            }
        }

        int left =0, right=sb.length()-1;
        while(left<right){
            if(sb.charAt(left++)!=sb.charAt(right--)) return false;
        }

        return true;
    }
}